pkg/client.go/reader.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	admin "google.golang.org/api/admin/directory/v1"
+	"google.golang.org/api/cloudidentity/v1"
+
+	"github.com/abcxyz/team-link/pkg/googlegroups"
+	"github.com/abcxyz/team-link/pkg/groupsync"
+)
+
+// TODO: take token to authenticate.
+// The token should be generated by WIF (google-github-actions/auth)
+// Currently we don't have a github action set up, we will use default
+// login credentials during dev phase.
+func NewGoogleGroupsReader(ctx context.Context) (groupsync.GroupReader, error) {
+	cs, err := cloudidentity.NewService(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create cloudidentity service: %w", err)
+	}
+	as, err := admin.NewService(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create admin service: %w", err)
+	}
+	return googlegroups.NewGroupReader(cs, as), nil
+}
+
+// NewReader creates a GroupReader base on provided source type.
+func NewReader(ctx context.Context, source string) (groupsync.GroupReader, error) {
+	if source == "GOOGLEGROUPS" {
```

Comment:
can we use `tltypes.SystemTypeGoogleGroups` here as well?

Reply:
oh, missed this one here. Updated.